### ISSUE TITLE
Can't install clippy with cargo: pattern `Catch(_)` not covered

### ISSUE THREAD

**Issue by atodorov** [Manishearth/rust-clippy#1622]
```
$ rustup show
Default host: x86_64-unknown-linux-gnu

nightly-x86_64-unknown-linux-gnu (default)
rustc 1.17.0-nightly (6f10e2f63 2017-03-14)

$ cargo install clippy
    Updating registry `https://github.com/rust-lang/crates.io-index`
  Installing clippy v0.0.119
   Compiling unicode-xid v0.0.4
   Compiling quine-mc_cluskey v0.2.4
   Compiling itoa v0.3.1
   Compiling quote v0.3.15
   Compiling synom v0.11.3
   Compiling rustc-serialize v0.3.22
   Compiling regex-syntax v0.4.0
   Compiling syn v0.11.9
   Compiling dtoa v0.4.1
   Compiling serde v0.9.11
   Compiling toml v0.2.1
   Compiling num-traits v0.1.37
   Compiling semver-parser v0.7.0
   Compiling serde_json v0.9.9
   Compiling serde_codegen_internals v0.14.1
   Compiling semver v0.6.0
   Compiling matches v0.1.4
   Compiling unicode-normalization v0.1.4
   Compiling clippy_lints v0.0.119
error[E0004]: non-exhaustive patterns: `Catch(_)` not covered
  --> /home/atodorov/.cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.119/src/utils/sugg.rs:90:15
   |
90 |         match expr.node {
   |               ^^^^^^^^^ pattern `Catch(_)` not covered

error: aborting due to previous error

error: Could not compile `clippy_lints`.
Build failed, waiting for other jobs to finish...
error: failed to compile `clippy v0.0.119`, intermediate artifacts can be found at `/tmp/cargo-install.V6QIugXKLAhp`

Caused by:
  build failed

```

**Comment by Arnavion** [comment]
(From https://github.com/rust-lang/rust/pull/39921 )

**Comment by llogiq** [comment]
A fix is pushed, but not yet published.

**Comment by oli-obk** [comment]
The fix has been published as 0.0.120

**Comment by oli-obk** [comment]
Sorry about the delay
